### PR TITLE
Clean up `metabase.query-processor-test.timezones-test/filter-datetime-by-date-in-timezone-test` 

### PR DESCRIPTION
**Pull request by camsaul**
Split off from a flaky test fix. Split a mega-test into three smaller tests, and other misc cleanup

### CODE REVIEW

File: modules/drivers/presto-jdbc/src/metabase/driver/presto_jdbc.clj [hunk 1]
```diff
@@ -56,7 +56,7 @@
 
 ;;; Presto API helpers
 
-(def presto-type->base-type
+(def ^:private ^{:arglists '([column-type])} presto-type->base-type
```
Comment (camsaul) [inline modules/drivers/presto-jdbc/src/metabase/driver/presto_jdbc.clj:59]: Not strictly related but a lot of this stuff should have been `^:private` so I fixed it

File: modules/drivers/presto-jdbc/src/metabase/driver/presto_jdbc.clj [hunk 2]
```diff
@@ -613,6 +613,12 @@
                                  :database-position idx}))
                  (jdbc/reducible-query {:connection conn} sql))})))
 
+;;; The Presto JDBC driver DOES NOT support the `.getImportedKeys` method so just return `nil` here so the `:sql-jdbc`
```
Comment (camsaul) [inline modules/drivers/presto-jdbc/src/metabase/driver/presto_jdbc.clj:616]: This fixes `:presto-jdbc` when running tests locally from the REPL

File: test/metabase/query_processor_test/timezones_test.clj [hunk 3]
```diff
@@ -297,42 +297,56 @@
...
+                    (is (= (-> expected-datetime
+                               (u.date/with-time-zone-same-instant timezone)
+                               t/offset-date-time)
+                           (some-> results
+                                   mt/first-row
+                                   first
+                                   (u.date/parse nil)
+                                   t/offset-date-time)))))))))))))
+
+(deftest filter-datetime-by-date-in-timezone-relative-to-days-since-test
```
Comment (camsaul) [inline test/metabase/query_processor_test/timezones_test.clj:324]: I split this one mega-test into three separate tests. There's no real reason it should all be one test, unless there is a lot of shared setup state or something it's probably better to have finer-grained tests. At least it's easier to debug

**Comment by camsaul** [comment]
This is actually failing for `:mysql` now too in the tests for this PR, which means we can utilize the new debugging info here.

https://github.com/metabase/metabase/actions/runs/4227507489/jobs/7342059102

```
FAIL in metabase.query-processor-test.timezones-test/filter-datetime-by-date-in-timezone-relative-to-current-date-test (timezones_test.clj:315)

:mysql Relative to current date using dataset-definition dataset
 
Setting :report-timezone = "UTC"
 
Native Query =
SELECT
  `relative_filter`.`created` AS `created`
FROM
  `relative_filter`
WHERE
  (
    `relative_filter`.`created` >= STR_TO_DATE(
      DATE_FORMAT(CAST(NOW(6) AS datetime), '%Y-%m-%d %H:%i'),
      '%Y-%m-%d %H:%i'
    )
  )
  AND (
    `relative_filter`.`created` < STR_TO_DATE(
      DATE_FORMAT(
        CAST(DATE_ADD(NOW(6), INTERVAL 1 minute) AS datetime),
        '%Y-%m-%d %H:%i'
      ),
      '%Y-%m-%d %H:%i'
    )
  )
LIMIT
  1048575

{:params nil}


expected: #<java.time.OffsetDateTime@77665125 2023-02-20T22:59:59Z>
  actual: nil
```

File: src/metabase/driver/sql_jdbc/sync/describe_table.clj [hunk 4]
```diff
@@ -229,15 +229,15 @@
...
-                                          {:fk-column-name   (.getString rs "FKCOLUMN_NAME")
-                                           :dest-table       {:name   (.getString rs "PKTABLE_NAME")
-                                                              :schema (.getString rs "PKTABLE_SCHEM")}
-                                           :dest-column-name (.getString rs "PKCOLUMN_NAME")})))))
+                                           (fn [^ResultSet rs]
+                                             (fn []
+                                               {:fk-column-name   (.getString rs "FKCOLUMN_NAME")
+                                                :dest-table       {:name   (.getString rs "PKTABLE_NAME")
+                                                                   :schema (.getString rs "PKTABLE_SCHEM")}
+                                                :dest-column-name (.getString rs "PKCOLUMN_NAME")})))))
```
Comment (camsaul) [inline src/metabase/driver/sql_jdbc/sync/describe_table.clj:237]: Fix indentation

File: test/metabase/actions/test_util.clj [hunk 5]
```diff
@@ -93,14 +93,14 @@
 (defmacro with-actions-test-data
   "Sets the current dataset to a freshly-loaded copy of [[defs/test-data]] that only includes the `categories` table
   that gets destroyed at the conclusion of `body`. Use this to test destructive actions that may modify the data."
-  {:style/indent 0}
+  {:style/indent :defn}
```
Comment (camsaul) [inline test/metabase/actions/test_util.clj:96]: Indentation spec for these macros was off

**Review by calherries — APPROVED** [review]

**Review by metamben — APPROVED** [review]

**Pull request merged**